activerecord/lib/active_record/connection_adapters/postgresql/oid/interval.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/duration'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Interval < Type::Value # :nodoc:
+          def type
+            :interval
+          end
+
+          def type_cast_from_database(value)
+            if value.kind_of? ::String
```

Comment:
Do we expect values other than a string or `nil` here? I'd rather not let those slip through. What do you think of changing to this instead?

```ruby
def type_cast_from_database(value)
  return if value.nil?
  ::ActiveSupport::Duration.parse(value)
end
```